### PR TITLE
Update font_manager to only use registry on Win

### LINKED ISSUE #22859: [Bug]: findSystemFonts should not look in subdirectories of C:\Windows\Fonts\
### Bug summary

findSystemFonts takes the font in the directory: C:\Windows\Fonts\Deleted

It should not take them. They are supposedly deleted. It is better to ignore them.

### Code for reproduction

```python
matplotlib.font_manager.findSystemFonts
```

### Actual outcome

---

### Expected outcome

---

### Additional information

_No response_

### Operating system

Windows

### Matplotlib Version

3.5.1

### Matplotlib Backend

_No response_

### Python version

_No response_

### Jupyter version

_No response_

### Installation

pip

### PR DESCRIPTION
**Pull request by almarklein**
* Follow-up on https://github.com/matplotlib/matplotlib/pull/24641, which introduces a bug (fonts are added regardless of their extension).
* Undoes https://github.com/matplotlib/matplotlib/pull/22909 and implements another way to fix https://github.com/matplotlib/matplotlib/issues/22859: don't scrape dirs with `list_fonts` on Windows.

### CODE REVIEW

**Review by anntzer — APPROVED** [review]

**Comment by tacaswell** [comment]
xref to https://github.com/matplotlib/matplotlib/issues/24001

**Comment by anntzer** [comment]
I think that this or something similar needs to make it into 3.7 to fix #24641 being partially incorrect; or #24641 needs to be temporarily reverted for 3.7.

**Comment by tacaswell** [comment]
```
____________________________ test_user_fonts_win32 ____________________________
[gw1] win32 -- Python 3.10.8 C:\hostedtoolcache\windows\Python\3.10.8\x64\python.exe

    @pytest.mark.skipif(sys.platform != 'win32', reason='Windows only')
    def test_user_fonts_win32():
        if not (os.environ.get('APPVEYOR') or os.environ.get('TF_BUILD')):
            pytest.xfail("This test should only run on CI (appveyor or azure) "
                         "as the developer's font directory should remain "
                         "unchanged.")
    
        font_test_file = 'mpltest.ttf'
    
        # Precondition: the test font should not be available
        fonts = findSystemFonts()
        if any(font_test_file in font for font in fonts):
            pytest.skip(f'{font_test_file} already exists in system fonts')
    
        user_fonts_dir = MSUserFontDirectories[0]
    
        # Make sure that the user font directory exists (this is probably not the
        # case on Windows versions < 1809)
        os.makedirs(user_fonts_dir)
    
        # Copy the test font to the user font directory
        shutil.copy(Path(__file__).parent / font_test_file, user_fonts_dir)
    
        # Now, the font should be available
        fonts = findSystemFonts()
>       assert any(font_test_file in font for font in fonts)
E       assert False
E        +  where False = any(<generator object test_user_fonts_win32.<locals>.<genexpr> at 0x000002364CB2A8F0>)
```
https://dev.azure.com/matplotlib/matplotlib/_build/results?buildId=30184&view=logs&j=595c9a33-3ac5-58f0-6ce6-44c63ffdfe86&t=6405dc77-f20a-51d5-22e4-a611ccc41f2e&l=182

The failure is definitely caused by this change

File: lib/matplotlib/font_manager.py [hunk 1]
```diff
@@ -275,7 +270,7 @@ def findSystemFonts(fontpaths=None, fontext='ttf'):
     if fontpaths is None:
         if sys.platform == 'win32':
             installed_fonts = _get_win32_installed_fonts()
-            fontpaths = MSUserFontDirectories + [win32FontDirectory()]
+            fontpaths = []
```
Comment (tacaswell) [inline lib/matplotlib/font_manager.py:273]:
```suggestion
            fontpaths = MSUserFontDirectories
```

Might work, but seems against the spirit of this change....

**Comment by QuLogic** [comment]
> Also, I dunno if this is sensible but I modified the test to run it on my local machine, and I'm now getting the same failure, w/ the same message:

Ah, are you saying the test is wrong? I think we maybe should be calling `AddFontResourceEx` for that test instead of directly copying the file.

**Comment by QuLogic** [comment]
I guess what I would like to confirm is; if you use Explorer to copy a new font there, does it get added to the registry? Does it only do so if you use the Install button on the font viewer? What about if you have a Python script that copies the font there; does the registry update automatically, or do you have to open the directory in Explorer to refresh it, or does nothing update the registry?

**Comment by tacaswell** [comment]
Per discussion on this weeks dev call:


 - go with windows registry only
 - xfail the test
     - defer questions of how to test this to later
     - do we want to change the registry as part of the test?
     - opt in to running locally?
     - flag to only run an CI + windows?
 - need a manual test that this works 
     - download a new font
     - use windows way to install it
     - remove mpl's font cache
     - verify the new font can be found
 - add behavior change note

**Review by tacaswell — APPROVED** [review]
but I wrote the deprecation note so this review should be taken with a grain of salt.

**Comment by tacaswell** [comment]
@story645 Can you please confirm that this works as expected for fonts installed as Windows expects and merge if you are satisfied?

File: doc/api/next_api_changes/behavior/24655-AK.rst [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+to discover fonts, however this lead to a number of undesirable behaviors
+including finding deleted fonts.  Now Matplotlib will only find fonts that are
+known to the Windows.
+
+This means that any user installed fonts must got through the Windows font
+installer rather than simply being copied to the correct folder.
+
+This only impacts the set of fonts Matplotlib will consider when using
+`matplotlib.font_manager.find_font`, directly pass the path to a font to use an
+arbitrary font (see :doc:`/gallery/text_labels_and_annotations/fonts_file`).
```
Comment (QuLogic) [inline doc/api/next_api_changes/behavior/24655-AK.rst:14]:
```suggestion
`matplotlib.font_manager.findfont`, directly pass the path to a font to use an
arbitrary font (see :doc:`/gallery/text_labels_and_annotations/font_file`).
```

File: doc/api/next_api_changes/behavior/24655-AK.rst [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+to discover fonts, however this lead to a number of undesirable behaviors
+including finding deleted fonts. Now Matplotlib will only find fonts that are
+known to the Windows registry.
+
+This means that any user installed fonts must go through the Windows font
+installer rather than simply being copied to the correct folder.
+
+This only impacts the set of fonts Matplotlib will consider when using
+`matplotlib.font_manager.findfont`. To use an arbitrary font, directly pass the
+path to a font to use an arbitrary font (see
```
[the remark this answers is not in the document]
Comment (QuLogic) [reply doc/api/next_api_changes/behavior/24655-AK.rst]: No, I misread your suggestion.

**Pull request merged**